Specs/Scene/ExpressionSpec.js

```diff
@@ -681,6 +686,68 @@ defineSuite([
         expect(expression.evaluate(feature)).toEqual(true);
     });
 
+    it('constructs regex', function() {
+        var expression = new Expression(new MockStyleEngine(), 'RegExp("a")');
+        expect(expression.evaluate(undefined)).toEqual(/a/);
+
+        expression = new Expression(new MockStyleEngine(), 'RegExp("\\w")');
+        expect(expression.evaluate(undefined)).toEqual(/\w/);
+
+        expression = new Expression(new MockStyleEngine(), 'RegExp(1)');
+        expect(expression.evaluate(undefined)).toEqual(/1/);
+
+        expression = new Expression(new MockStyleEngine(), 'RegExp(true)');
+        expect(expression.evaluate(undefined)).toEqual(/true/);
+
+        expression = new Expression(new MockStyleEngine(), 'RegExp()');
+        expect(expression.evaluate(undefined)).toEqual(/(?:)/);
+
+    });
+
+    it ('constructs regex with flags', function() {
+        var expression = new Expression(new MockStyleEngine(), 'RegExp("a", "i")');
+        expect(expression.evaluate(undefined)).toEqual(/a/i);
+
+        expression = new Expression(new MockStyleEngine(), 'RegExp("a", "mg")');
+        expect(expression.evaluate(undefined)).toEqual(/a/mg);
+    });
+
+    it('throws if regex constructor does not have literal as argument', function() {
+        expect(function() {
+            return new Expression(new MockStyleEngine(), 'RegExp(1+1)');
+        }).toThrowDeveloperError();
+    });
+
+    it('throws if regex constructor has invalid flags', function() {
+        expect(function() {
+            return new Expression(new MockStyleEngine(), 'RegExp("a", "q")');
+        }).toThrowDeveloperError();
+    });
+
+    it('evaluates regex matches operator', function() {
```

Comment:
We don't need to overdo it since we can trust the JavaScript RegExp engine, but I think this needs a couple more non-trivial tests with more complicated regular expressions and compound expressions.